sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -154,18 +202,8 @@ void gnrc_ipv6_nib_init_iface(gnrc_netif_t *netif)
     }
     _add_static_lladdr(netif);
     _auto_configure_addr(netif, &ipv6_addr_link_local_prefix, 64U);
-    if (!(gnrc_netif_is_rtr_adv(netif)) ||
-        (gnrc_netif_is_6ln(netif) && !gnrc_netif_is_6lbr(netif))) {
-        uint32_t next_rs_time = random_uint32_range(0, NDP_MAX_RS_MS_DELAY);
 
-        _evtimer_add(netif, GNRC_IPV6_NIB_SEARCH_RTR, &netif->ipv6.search_rtr,
-                     next_rs_time);
-    }
-#if IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_ROUTER)
-    else {
-        _handle_snd_mc_ra(netif);
-    }
-#endif  /* CONFIG_GNRC_IPV6_NIB_ROUTER */
+    gnrc_ipv6_nib_iface_up(netif);
```

Comment:
Should this be guarded to only be executed if the interface is up (or checking the if/up down state is not supported)? Similar to lwip?

I think on STM32 Ethernet GNRC doesn't do DAD because it sends it out prior to the link actually being up (resulting in the solicitation not actually being sent) and GNRC then assuming no duplicates are detected, even if one may be there.

Reply:
The problem there is that only a fraction of the drivers that implement `NETOPT_LINK` also provide a `NETDEV_EVENT_LINK_UP` event. So those would then never do DAD / send router solicitations.

Reply:
Then maybe we poll `NETOPT_LINK` and wait for `NETOPT_ENABLE`. And in the handler of `NETDEV_EVENT_LINK_UP` we return immediately if the link is up already.
Not nice, but should work.

Reply:
Like inside the netif thread?

Reply:
Yes it could have been inside the netif thread but I noticed that this does not solve the double execution of `gnrc_ipv6_nib_iface_up()`.

> And in the handler of NETDEV_EVENT_LINK_UP we return immediately if the link is up already.

This did not make sense because `gnrc_ipv6_nib_iface_up()` does not actively bring up the interface.

So new proposal: `gnrc_ipv6_nib_init_iface()` does not call `gnrc_ipv6_nib_iface_up()`, but every interface driver must at some point call `netdev->event_callback(netdev, NETDEV_EVENT_LINK_UP);` and may it be at the end of it´s `init()` function, if there is no interrupt for that.